### DISCUSSION TITLE
Unbound vs AdGuard Home internal cache

### DISCUSSION THREAD

**Discussion by vdias** [trinib/Adguard-Wireguard-Unbound-Cloudflare#68] [General]
One quick question why do we need unbound if adguard home already have a cache option?

**Comment by trinib** [comment]
@vdias 

Unbound and AdGuard Home serve different purposes, even though they both offer a caching option.

AdGuard Home is primarily a DNS server that filters and blocks requests to known malicious or unwanted domains. It has a caching option that stores DNS responses for a certain amount of time, reducing the time it takes to retrieve the same data again in the future. This caching feature can help speed up DNS resolution and reduce network traffic by reducing the number of queries that need to be sent to other DNS servers.

Unbound, on the other hand, is a full-featured recursive DNS server that can act as a resolver for all DNS queries, not just those related to ad-blocking. It also has a caching feature that stores responses for a certain amount of time, reducing the time it takes to retrieve the same data again in the future. In addition to caching, Unbound has features like DNSSEC validation and support for DNS over TLS, making it a more comprehensive solution for DNS resolution.

So while AdGuard Home's caching option can help speed up DNS resolution and reduce network traffic, Unbound's caching and additional features make it a more powerful DNS server overall.

**Reply by KaKi87** [reply]
What's the practical difference between *a caching option that stores DNS responses* and *a full-featured recursive DNS server* though ? Doesn't even the second one have to get data from somewhere anyway ? Thanks

**Comment by vdias** [comment]
Adguard Home, as outbound DNSSEC and DNS over TLS... thats why i'm asking about it... both have securiyty settings and local cache... why do you guys recommend using both together?
